### PR TITLE
Gradient Descent (R. F.)

### PR DESCRIPTION
**Pull request by Rfarzad**
This work aims to introduce and compare three kinds of Gradient Descent methods (BGD or Batch Gradient Descent, mBGD or mini-Batch Gradient Descent, and SGD or Stochastic Gradient Descent). The corresponding formulation and algorithms are clearly presented, the effect of different learning rates is depicted, and the difference between the run-time for large and small datasets is demonstrated.

### CODE REVIEW

**Comment by Rfarzad** [comment]
@adowling2
